enhancements/microshift/kubernetes-for-device-edge.md

```diff
@@ -0,0 +1,520 @@
+
+A device administrator manages the lifecycle of a device running the
+applications.
+
+1. As an application developer, I can design my application deployment
+   so that it works with MicroShift and all other topologies of
+   OpenShift in a consistent way.
+2. As an application deployer, I can package MicroShift, the
+   applications, and any other platform components into a "system image" that
+   is written onto many devices by the manufacturer.
+3. As an application deployer, I can package MicroShift, the
+   applications, and any other platform components into a "system image" that
+   I install myself.
+4. As a device administrator, I can release an updated system image over the
+   network to upgrade devices already in the field using minimal
+   network bandwidth.
+5. As a device administrator, I can specify application-specific logic
+   with which the device will verify a successful upgrade and roll
+   back the upgrade automatically if verification fails.
+6. As a device administrator, I have control over the operating system
+   running under MicroShift and can install kernel drivers, other
+   applications, agents, etc. easily.
+
+### Goals
+
+* Describe an approach to providing a version of Kubernetes compatible
+  with the OpenShift APIs that it includes and that meets the
+  constraints described in the Motivation section.
+* Identify the minimal set of APIs that will be included and the
+  components needed to support them.
+* Design a system that can work autonomously, without requiring
+  external orchestration tools for deployment.
+* Treat MicroShift as an application running on RHEL, with the minimum
+  privileges possible.
+* Design a system that can be configured by someone familiar with
+  Linux, without Kubernetes expertise.
+* Follow the [MicroShift design
+  principles](https://github.com/openshift/microshift/blob/main/docs/design.md)
+
+### Non-Goals
+
+* We are not trying to fit all of the features of a standalone
+  OpenShift deployment into the target hardware platforms.
+* We are not trying to make MicroShift self-managing, so no core
+  OpenShift operators will be included.
+* We do not plan to support multi-node clusters.
+* The details of the default CSI driver are left to a future
+  enhancement.
+* The details of the default CNI driver are left to a future
+  enhancement.
+* Use cases beyond industrial applications, such as developer
+  environments, are out of scope for now.
+* The design of MicroShift relies heavily on features of RHEL for Edge
+  and rpm-ostree. Those features are referenced here, but their
+  implementation is not described in detail.
+* The details of integrating with fleet orchestration tools such as
+  Multi-cluster Engine (MCE) or Advanced Cluster Manager (ACM) are
+  left to a future enhancement.
+* We do not want to push MicroShift use on RHEL versions where OCP is
+  not supported. We do have interest from users in RHEL 9, but will
+  not GA on RHEL 9 faster than OCP does.
```

Comment:
Yeah, we called this out because from a support standpoint we'd really rather let the testing of standalone clusters uncover issues that come in with the new OS.